### PR TITLE
SemanticsUpdateBuilder.updateNode(): make TextDirection non-nullable

### PR DESCRIPTION
**Pull request by bartekpacia**
This PR is a very tiny cleanup that attempts to make all params of `SemanticsUpdateBuilder.updateNode()` non-nullable.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: lib/ui/semantics.dart [hunk 1]
```diff
@@ -818,7 +818,7 @@ abstract class SemanticsUpdateBuilder {
     required String hint,
     required List<StringAttribute> hintAttributes,
     required String tooltip,
-    required TextDirection? textDirection,
+    required TextDirection textDirection,
```
Comment (chunhtai) [inline lib/ui/semantics.dart:821]: I am surprised this is not breaking the framework https://github.com/flutter/flutter/blob/28b81dc575da37247cb0719b17883b8db65560f6/packages/flutter/lib/src/semantics/semantics.dart#L2799
data.textDirection is nullable.

do you have another pr to migrate framework code?

File: lib/ui/semantics.dart [hunk 2]
```diff
@@ -927,7 +927,7 @@ base class _NativeSemanticsUpdateBuilder extends NativeFieldWrapperClass1 implem
       hint,
       hintAttributes,
       tooltip,
-      textDirection != null ? textDirection.index + 1 : 0,
+      textDirection.index + 1,
```
Comment (chunhtai) [inline lib/ui/semantics.dart:930]: what happen to the case where textDirection used to be null? I think we want to send 0 to the native side?

File: lib/ui/semantics.dart (hunk 1 above)
Comment (bartekpacia) [reply lib/ui/semantics.dart:821]: I am also surprised. I was suspecting that this PR will have red CI, and then I'd submit a separate framework PR.

File: lib/ui/semantics.dart (hunk 2 above)
Comment (bartekpacia) [reply lib/ui/semantics.dart:930]: If the native side has to handle TextDirection being null, and it's a valid use case, then I think this PR doesn't make sense and can be closed. Wdyt?

Comment (chunhtai) [reply lib/ui/semantics.dart:930]: I only take a quick glance we only enforce TextDirection to be non-null when there is and text data in the SemanticsData. I think this pr may not be the right move

Comment (bartekpacia) [reply lib/ui/semantics.dart:930]: I think a new test should be added, so when in the future someone tries to do the same PR as this one, that test would fail.

Comment (chunhtai) [reply lib/ui/semantics.dart:930]: This is weird I remember we used to have a framework smoke test, but it was gone from the ci

File: lib/ui/semantics.dart (hunk 1 above)
Comment (chunhtai) [reply lib/ui/semantics.dart:821]: It looks like they are disabled here https://github.com/flutter/engine/pull/50800

File: lib/ui/semantics.dart (hunk 2 above)
Comment (chunhtai) [reply lib/ui/semantics.dart:930]: It was disabled here https://github.com/flutter/engine/pull/50800

**Comment by bartekpacia** [comment]
Closing as per https://github.com/flutter/engine/pull/52421#discussion_r1584974986
